### PR TITLE
Implement CLI functionality

### PR DESCRIPTION
**Pull request by jmcph4**
# Description #
See #1. 

# Rationale #
See #1. 

# Acceptance Criteria #
 - [x] Invocations of `rainyday` fail and display usage information
 - [x] Invocations of `rainyday --help` succeed and display help information
 - [x] Invocations of `rainyday --version` succeed and display version information

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,3 +7,5 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+clap = "2.33.3"
```
Comment (alecdwm) [inline Cargo.toml]: Clap v3's [derive macro is pretty cool](https://github.com/clap-rs/clap#using-derive-macros) if you wanted to try it out

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,8 @@
+use clap::{App, Arg, ArgMatches};
+
 fn main() {
-    println!("Hello, world!");
+    let _matches: ArgMatches = App::new("rainyday")
+        .version("0.0.1")
```
Comment (alecdwm) [inline src/main.rs]: You can pull this from the cargo.toml like so:

`.version(crate_version!())`

[example here](https://github.com/clap-rs/clap/blob/HEAD/examples/09_auto_version.rs)

**Review by alecdwm — COMMENTED** [review]
Added some comments

File: src/main.rs (hunk 2 above)
Comment (jmcph4) [reply src/main.rs]: Very cool! See 1ea28dd.

File: Cargo.toml (hunk 1 above)
Comment (jmcph4) [reply Cargo.toml]: Noted, see 5eff499.

File: Cargo.toml [hunk 3]
```diff
@@ -7,3 +7,5 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+clap = { git = "https://github.com/clap-rs/clap/" }
```
Comment (alecdwm) [inline Cargo.toml]: They have a prerelease on crates.io!
`clap = "3.0.0-beta.2"`

**Review by alecdwm — COMMENTED** [review]
Looks great! One more comment

File: Cargo.toml (hunk 3 above)
Comment (jmcph4) [reply Cargo.toml]: Good catch! Fixed in 19cb46f.

**Review by alecdwm — APPROVED** [review]

**Pull request merged**